popx/transaction.go

```diff
@@ -74,3 +81,35 @@ func (s sqlxTxAdapter) Commit(ctx context.Context) error {
 func (s sqlxTxAdapter) Rollback(ctx context.Context) error {
 	return s.Tx.Rollback()
 }
+
+var (
+	TransactionRetries = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "ory_x_popx_cockroach_transaction_retries_total",
+		Help: "Counts the number of automatic CockroachDB transaction retries",
+	}, []string{"caller"})
+	_             = TransactionRetries.WithLabelValues(unknownCaller) // make sure the metric is always present
+	unknownCaller = "unknown"
+)
+
+func caller() string {
```

Comment:
There's really no way to tell if the number of skipped frames is "correct", since usage will vary. 8 is my best guess from common usage in kratos. We'll have to try and see in actual usage how many suboptimal labels we get.